core/trie/trie.go

```diff
@@ -195,8 +195,8 @@ func (t *Trie) Get(key *felt.Felt) (*felt.Felt, error) {
 	storageKey := t.feltToKey(key)
 	value, err := t.storage.Get(&storageKey)
 	if err != nil {
-		if errors.Is(err, db.ErrKeyNotFound) {
-			return &felt.Zero, nil
+		if errors.Is(err, db.ErrKeyNotFound) { // a non-existent key returns a zero value
+			return &felt.Zero, db.ErrKeyNotFound
```

Comment:
why do we need to pass `felt.Zero` in this case?

Reply:
As mentioned in the comment, a key that does not exists in the trie by default has a zero value

Reply:
yes, but I'm wondering how useful it would be if usual pattern is:
```
v, err := func()
if err != nil 
     return err
}
```
since we started to return `db.ErrKeyNotFound`  error